src/main/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilder.java

```diff
@@ -476,14 +472,18 @@ else if (STREAM_TYPES.contains(this.outType))
                 writer = createBCFWriter(outPath, outStreamFromFile);
                 break;
             case VCF_STREAM:
-                if (options.contains(Options.INDEX_ON_THE_FLY))
-                    throw new IllegalArgumentException("VCF index creation not supported for stream output.");
+                if (options.contains(Options.INDEX_ON_THE_FLY)) {
```

Comment:
to avoid code duplication this check can be done on line 433 above, after checking `STREAM_TYPES.contains(this.outType)`

Reply:
can you use htsjdk.samtools.util.IOUtil#isRegularPath(java.io.File)?

Reply:
The issue with this approach is that the user of this builder can set the output path directly. If this happens then there is no way to extract what or where the stream points to and consequently we can't reliably determine the output value here. The plan was to just to call htsjdk.samtools.util.IOUtil#isRegularPath(java.io.File) in gatk to sidestep this issue. Either way disabling index-on-the-fly would save users/tool authors the trouble of having their streams crash if outputting to an unindexable place like "/dev/null"